dds/DCPS/DataReaderImpl_T.h

```diff
@@ -975,6 +988,15 @@ namespace OpenDDS {
                              OpenDDS::DCPS::MarshalingType marshaling_type)
   {
     unique_ptr<MessageTypeWithAllocator> data(new (*data_allocator()) MessageTypeWithAllocator);
+
+    if (marshal_skip_serialize_) {
+      if(!MarshalTraitsType::from_message_block(static_cast<MessageType&>(*data), static_cast<ACE_Message_Block&>(*sample.sample_))) {
```

Comment:
```suggestion
      if (!MarshalTraitsType::from_message_block(static_cast<MessageType&>(*data), static_cast<ACE_Message_Block&>(*sample.sample_))) {
```
Are both of these casts necessary?